data_safe_haven/infrastructure/programs/sre/networking.py

```diff
@@ -2070,103 +1548,734 @@ def __init__(
+                protocol=network.SecurityRuleProtocol.TCP,
+                source_address_prefix=SREIpRanges.workspaces.prefix,
+                source_port_range="*",
+            ),
+            network.SecurityRuleArgs(
+                access=network.SecurityRuleAccess.ALLOW,
+                description="Allow outbound connections to user services databases.",
+                destination_address_prefix=SREIpRanges.user_services_databases.prefix,
+                destination_port_ranges=[Ports.MSSQL, Ports.POSTGRESQL],
+                direction=network.SecurityRuleDirection.OUTBOUND,
+                name="AllowUserServicesDatabasesOutbound",
+                priority=NetworkingPriorities.INTERNAL_SRE_USER_SERVICES_DATABASES,
+                protocol=network.SecurityRuleProtocol.TCP,
+                source_address_prefix=SREIpRanges.workspaces.prefix,
+                source_port_range="*",
+            ),
+            self.get_nsg_workspaces_nexus_outbound_rule_args(),
+            network.SecurityRuleArgs(
+                access=network.SecurityRuleAccess.ALLOW,
+                description="Allow outbound connections to apt proxy server.",
+                destination_address_prefix=SREIpRanges.apt_proxy_server.prefix,
+                destination_port_ranges=[Ports.LINUX_UPDATE],
+                direction=network.SecurityRuleDirection.OUTBOUND,
+                name="AllowAptProxyServerOutbound",
+                priority=NetworkingPriorities.INTERNAL_SRE_APT_PROXY_SERVER,
+                protocol=network.SecurityRuleProtocol.TCP,
+                source_address_prefix=SREIpRanges.workspaces.prefix,
+                source_port_range="*",
+            ),
+            network.SecurityRuleArgs(
+                access=network.SecurityRuleAccess.ALLOW,
+                description="Allow outbound configuration traffic over the internet.",
+                destination_address_prefix="Internet",
+                destination_port_range="*",
+                direction=network.SecurityRuleDirection.OUTBOUND,
+                name="AllowConfigurationInternetOutbound",
+                priority=NetworkingPriorities.EXTERNAL_INTERNET,
+                protocol=network.SecurityRuleProtocol.TCP,
+                source_address_prefix=SREIpRanges.workspaces.prefix,
+                source_port_range="*",
+            ),
+            network.SecurityRuleArgs(
+                access=network.SecurityRuleAccess.DENY,
+                description="Deny all other outbound traffic.",
+                destination_address_prefix="*",
+                destination_port_range="*",
+                direction=network.SecurityRuleDirection.OUTBOUND,
+                name="DenyAllOtherOutbound",
+                priority=NetworkingPriorities.ALL_OTHER,
+                protocol=network.SecurityRuleProtocol.ASTERISK,
+                source_address_prefix="*",
+                source_port_range="*",
+            ),
+        ]
+
+        return [
```

Comment:
Same as above. Why are we allowing None in this list? Especially, why is that then part of the return type, when I think it could just be an implementation detail.